integration-tests/modules/__tests__/product/store/index.spec.ts

```diff
@@ -0,0 +1,332 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  CreateProductDTO,
+  IPricingModuleService,
+  IProductModuleService,
+  ISalesChannelModuleService,
+  ProductDTO,
+  ProductVariantDTO,
+} from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  Modules,
+  ProductStatus,
+} from "@medusajs/utils"
+import { medusaIntegrationTestRunner } from "medusa-test-utils"
+import { createAdminUser } from "../../../../helpers/create-admin-user"
+import { createDefaultRuleTypes } from "../../../helpers/create-default-rule-types"
+import { createVariantPriceSet } from "../../../helpers/create-variant-price-set"
+
+jest.setTimeout(50000)
+
+const env = { MEDUSA_FF_MEDUSA_V2: true }
+const adminHeaders = { headers: { "x-medusa-access-token": "test_token" } }
+
+async function createProductsWithVariants(
+  productModule: IProductModuleService,
+  productsData: CreateProductDTO
+): Promise<[ProductDTO, ProductVariantDTO[]]> {
+  const { variants: variantsData, ...productData } = productsData
+
+  const [product] = await productModule.create([productData])
+  const variantsDataWithProductId = variantsData?.map((variantData) => {
+    return { ...variantData, product_id: product.id }
+  })
+
+  const variants = variantsDataWithProductId
+    ? await productModule.createVariants(variantsDataWithProductId)
+    : []
+
+  return [product, variants]
+}
+
+medusaIntegrationTestRunner({
+  env,
+  testSuite: ({ dbConnection, getContainer, api }) => {
+    describe("Store: Products API", () => {
+      let appContainer
+      let product
+      let product2
+      let product3
+      let product4
+      let variant
+      let variant2
+      let variant3
+      let variant4
+      let pricingModule: IPricingModuleService
```

Comment:
Ideally we should not directly access the modules from the e2e tests, we should treat them as a black box and rely on the HTTP endpoints. The only exception is if there is something that isn't accessible through the HTTP endpoints and we really need to validate it (eg. due to some background task), but that should be rare